apps/frontend/src/app/CreateOrder/CreateOrder.tsx

```diff
@@ -1,51 +1,127 @@
+  const [DeliveryPrice, setDeliveryPrice] = useState<string>('');
+
+  const deliveryRef = useRef<HTMLInputElement>(null);
+
   const [error, setError] = useState<string | null>(null);
   const history = useHistory();
 
   const onSubmit = (data: CreateOrderDto) => {
-    createOrder({ ...data, orderAt: new Date().toISOString() })
-      .then((params: OrderModel) => history.push(`/`))
+    const newPrice = data.shippingCents
+      ? +formatter.format(data.shippingCents)
+      : -1;
+    createOrder({
+      ...data,
+      orderAt: new Date().toISOString(),
+      shippingCents: newPrice * 100,
+    })
+      .then((params: OrderModel) => {
+        history.push(`order/` + params.id);
+      })
       .catch((err: Error) => setError(err.message));
   };
 
-  const clearError = () => {
-    setError(null);
-  };
+  const handleChangeRestaurant = useCallback(
+    (e: ChangeEvent<HTMLInputElement>) => {
+      setRestaurant(e.currentTarget.value);
+    },
+    [setRestaurant]
+  );
+
+  const handleChangeMenu = useCallback(
+    (e: ChangeEvent<HTMLInputElement>) => {
+      setMenu(e.currentTarget.value);
+      setIsValid(validUrl(e.currentTarget.value));
+      setIsValidInput(validUrl(e.currentTarget.value));
+    },
+    [setMenu]
+  );
+
+  const handleChangeDeliveryPrice = useCallback(
+    (e: ChangeEvent<HTMLInputElement>) => {
+      setDeliveryPrice(e.currentTarget.value);
+    },
+    [setDeliveryPrice]
+  );
 
   return (
-    <div className="bg-gray-100 w-full h-screen">
-      <div className="container flex justify-center">
-        <div className=" bg-white p-6 rounded">
-          <div>{error}</div>
-          <form onSubmit={handleSubmit(onSubmit)}>
-            <input
-              className="w-full p-4 border border-gray-400 rounded mb-2"
-              {...register('from')}
-              placeholder="Restaurant name"
-            />
-            <input
-              className="w-full p-4 border border-gray-400 rounded mb-2"
-              {...register('menuUrl')}
-              placeholder="Menu URL"
-            />
-            <button
-              className="p-4 px-6 bg-blue-400 text-white rounded"
-              type="submit"
-            >
-              Create order
-            </button>
-          </form>
-        </div>
-      </div>
+    <div>
+      <Paper elevation={3} className="formSquare">
+        {error && <Alert severity="error">Request Failed!</Alert>}
+
+        <form className="form" onSubmit={handleSubmit(onSubmit)}>
+          <TextField
+            autoFocus
+            margin="normal"
+            label="Restaurant"
+            value={restaurant}
+            inputProps={{
+              fullWidth: true,
+            }}
+            {...register('from')}
+            onChange={handleChangeRestaurant}
+            type="text"
+            required
+          />
+          <TextField
+            margin="normal"
+            error={isValidInput}
```

Comment:
this basically means that `error === true` when input is valid `isValidInput === true` 
something is wrong with names